Remove legacy checkout checkbox from settings page

malithsen opened the pull request:
<!--
Did I add a title? A descriptive, yet concise, title.
-->

<!--
Issue: Link to the GitHub issue this PR addresses (if appropriate).
-->

Fixes STRIPE-535

## Changes proposed in this Pull Request:

- Remove the UI checkbox to enable Legacy Checkout.
- Remove Legacy Checkout e2e tests as there's no way to setup tests without the checkbox.

Not removed:
- [Tracks events](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/0a5aa742f2e2f7d9fc1776f9b5a8c7d092b898f0/client/settings/upe-toggle/provider.js#L8)
- [System report labels](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/0a5aa742f2e2f7d9fc1776f9b5a8c7d092b898f0/includes/class-wc-stripe-status.php#L116)
- References to legacy checkout throughout the codebase.

We'd want to keep them until we remove the filter introduced in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4385
<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for the PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

## Testing instructions
1. Go to Stripe settings `YOUR_TEST_SITE/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`.
2. Make sure the checkbox to enable Legacy Checkout is not visible.
3. (Optional exploratory test) Make sure there's no other option in the UI to enable Legacy checkout
 
<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Tested on mobile (or does not apply)

### Changelog entry

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

malithsen committed:
Remove ExperimentalFeatures component from advanced settings section

This component only contains the Legacy Checkout checkbox

malithsen committed:
Add changelog entries

malithsen committed:
Adjust a JS test

malithsen committed:
Remove legacy checkout e2e tests

malithsen committed:
Remove perf files

malithsen committed:
Merge branch 'develop' into update/remove-legacy-toggle

Mayisha commented:
Let's remove [this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/update/remove-legacy-toggle/includes/admin/class-wc-rest-stripe-settings-controller.php#L328) from the update function also. Removing only from the [update function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/update/remove-legacy-toggle/includes/admin/class-wc-rest-stripe-settings-controller.php#L302) is sufficient as we still want the value in GET response.

malithsen committed:
Stop saving the upe_enabled flag from settings

This flag is now always set to true. Setting it to false is no longer
possible via UI.

malithsen committed:
Merge branch 'develop' into update/remove-legacy-toggle

malithsen commented:
> Let's remove [this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/update/remove-legacy-toggle/includes/admin/class-wc-rest-stripe-settings-controller.php#L328) from the update function also. Removing only from the [update function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/update/remove-legacy-toggle/includes/admin/class-wc-rest-stripe-settings-controller.php#L302) is sufficient as we still want the value in GET response

@Mayisha Thanks, that makes sense. Updated in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4422/commits/3915e76250ec291f92a906f13931d78861a7978c

Mayisha reviewed (approved):
Looks good to me 👍 

✅ Legacy checkout checkbox is not displayed anymore on the settings page.
✅ Updating and saving the settings works without any errors.

annemirasol commented:
> > Let's remove [this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/update/remove-legacy-toggle/includes/admin/class-wc-rest-stripe-settings-controller.php#L328) from the update function also. Removing only from the [update function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/update/remove-legacy-toggle/includes/admin/class-wc-rest-stripe-settings-controller.php#L302) is sufficient as we still want the value in GET response
> 
> @Mayisha Thanks, that makes sense. Updated in [3915e76](https://github.com/woocommerce/woocommerce-gateway-stripe/commit/3915e76250ec291f92a906f13931d78861a7978c)

Should we also remove the method `update_is_upe_enabled` itself? It is private and no longer being used anywhere else.

malithsen committed:
Remove unused private method

malithsen commented:
> Should we also remove the method update_is_upe_enabled itself? It is private and no longer being used anywhere else.

@annemirasol Agreed. Removed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4422/commits/dd7bb0770969db7eac4453ee75e333dbed8225af

annemirasol reviewed (approved):
No regressions observed, LGTM!
